src/fragments/lib/auth/common/mfa/flows.mdx

```diff
@@ -0,0 +1,214 @@
+
+Refer to [SMS sandbox](/lib/auth/sms_flows#sandbox-mode).
+
+For information about Amazon SNS pricing, see [Worldwide SMS Pricing](https://aws.amazon.com/sns/sms-pricing/). 
+</Callout>
+
+<BlockSwitcher>
+
+<Block name="New Project">
+
+Run `amplify add auth` to create a new Cognito Auth resource, and follow the prompts below depending on how you want to integrate MFA into your flow.
+
+Turning MFA "ON" will make it required for all users, while "Optional" will make it available to enable on a per-user basis.
+
+import all2 from "/src/fragments/lib/auth/common/mfa/add_mfa.mdx";
+
+<Fragments fragments={{all: all2}} />
+
+</Block>
+
+<Block name="Existing Auth">
+
+Run `amplify update auth` and follow the prompts as guided below.
+
+import all5 from "/src/fragments/lib/auth/common/mfa/update_mfa.mdx";
+
+<Fragments fragments={{all: all5}} />
+
+</Block>
+
+</BlockSwitcher>
+
+## Multi-factor authentication with SMS
+
+<Callout>
+
+<b>AWS Cognito Limitation</b>
+
+The `fetchMFAPreference` API will only return enabled and preferred MFA methods, if the preferences were set using the `updateMFAPreference` API. If the preferences were not set using the `updateMFAPreference` API, the MFA preferences will return `nil` for enabled and preferred MFA methods. 
+
+Therefore it is recommended to use the `updateMFAPreference` API to set the MFA preferences for the user, if the user decides to set up any MFA methods. An example use case would be that when a user successfully sets up TOTP using the `setupTOTP` and `verifyTOTPSetup` API, `updateMFAPreference` should be invoked to set the MFA preferences for the user i.e. marking TOTP as enabled.
+
+</Callout>
+
+### Adding phone number during Sign Up
+
+Sign up users normally with the chosen `Username` type and password. `"phone_number"` will be required in the `userAttributes` if SMS MFA is ON, or manually enabled for the user. Note that if the username attribute is the `phone_number`, then adding a separate attribute for phone number is not required during sign up.
+
+import flutter6 from "/src/fragments/lib/auth/flutter/sms/sign_up.mdx";
+
+<Fragments fragments={{flutter: flutter6}} />
+
+import iosSignUp from "/src/fragments/lib/auth/ios/sms/sign_up.mdx";
+
+<Fragments fragments={{ios: iosSignUp}} />
+
+Verification of user accounts is done via the `confirmSignUp` method with the OTP sent to their phone or email.
+
+import flutter7 from "/src/fragments/lib/auth/flutter/sms/confirm_sign_up.mdx";
+
+<Fragments fragments={{flutter: flutter7}} />
+
+import iosConfirmSignUp from "/src/fragments/lib/auth/ios/sms/confirm_sign_up.mdx";
+
+<Fragments fragments={{ios: iosConfirmSignUp}} />
+
+### Handling SMS MFA challenge during Sign In
+
+Sign in users normally with the chosen `Username` type and password.
```

Comment:
After a user signs in, if they have MFA enabled for their account, a challenge will be returned that you would need to call the `confirmSignIn` API where the user provides their confirmation code sent to their phone number.